src/canvas/renderCanvas.js

```diff
@@ -99,206 +96,290 @@ function renderReactMarksAndMeasures(
+    //       showGroupHighlight: false,
+    //       showHoverHighlight,
+    //       state,
+    //     });
+    //   });
+
+    //   // Draw the hovered and/or selected items on top so they stand out.
+    //   // This is helpful if there are multiple (overlapping) items close to each other.
+    //   // if (hoveredEvent !== null && hoveredEvent.event !== null) {
+    //   //   renderSingleReactMarkOrMeasure({
+    //   //     baseY,
+    //   //     canvasWidth,
+    //   //     context,
+    //   //     eventOrMeasure: hoveredEvent.event,
+    //   //     showGroupHighlight: false,
+    //   //     showHoverHighlight: true,
+    //   //     state,
+    //   //   });
+    //   // }
+
+    //   baseY += REACT_EVENT_SIZE + REACT_GUTTER_SIZE;
+    // }
+
+    data.measures
+      // TODO: Optimization: precompute this so that we don't filter this array |lanesToRender| times
+      .filter(measure => measure.lanes.includes(lane))
+      .forEach(measure => {
+        const showHoverHighlight =
+          hoveredEvent && hoveredEvent.measure === measure;
+        const showGroupHighlight =
+          hoveredEvent &&
+          hoveredEvent.measure !== null &&
+          hoveredEvent.measure.batchUID === measure.batchUID;
+        renderSingleReactMeasure(
           context,
-          eventOrMeasure: event,
-          showGroupHighlight: false,
-          showHoverHighlight,
-          priorityIndex,
           state,
-        });
-      });
-
-      // Draw the hovered and/or selected items on top so they stand out.
-      // This is helpful if there are multiple (overlapping) items close to each other.
-      if (hoveredEvent !== null && hoveredEvent.event !== null) {
-        renderSingleReactMarkOrMeasure({
-          baseY,
+          measure,
           canvasWidth,
-          context,
-          eventOrMeasure: hoveredEvent.event,
-          showGroupHighlight: false,
-          showHoverHighlight: true,
-          priorityIndex: hoveredEvent.priorityIndex,
-          state,
-        });
-      }
-
-      baseY += REACT_EVENT_SIZE + REACT_GUTTER_SIZE;
-    }
-
-    currentPriority.measures.forEach(measure => {
-      const showHoverHighlight =
-        hoveredEvent && hoveredEvent.measure === measure;
-      const showGroupHighlight =
-        hoveredEvent &&
-        hoveredEvent.measure !== null &&
-        hoveredEvent.measure.batchUID === measure.batchUID;
-      renderSingleReactMarkOrMeasure({
-        baseY,
-        canvasWidth,
-        context,
-        eventOrMeasure: measure,
-        priorityIndex,
-        showGroupHighlight,
-        showHoverHighlight,
-        state,
+          baseY,
+          showGroupHighlight,
+          showHoverHighlight,
+        );
       });
-    });
 
-    priorityMinY += getPriorityHeight(data, priority);
+    laneMinY += getLaneHeight(data, lane);
   });
+
+  return laneMinY;
 }
 
-function renderSingleReactMarkOrMeasure({
-  baseY,
-  canvasWidth,
+function renderSingleReactMeasure(
```

Comment:
I wonder if these big functions would be better off using an object argument with destructuring to simulate named arguments?

For long args lists, named arguments are more readable than positional at the call site, and since these args are mostly primitive data types, it's easy to pass the wrong argument at wrong position and still pass flow check

Reply:
Also, IMO these functions would be easier to grok if we explicitly stated the input/output types, but I'm not married to that idea.

Reply:
Yeah, these suggestions make sense. I'll do these for all the render* functions in a PR on top of 3/n so that we cover all the functions added after this PR and also so that we don't get too many rebase conflicts

Reply:
Actually, this doesn't seem important enough to warrant the confusion of stacked PR shuffling, so I'll add it as a task in #30 to be done in another PR at the top of the stack